pkg/scheduler/core/division_algorithm.go

```diff
@@ -91,6 +82,35 @@ func divideReplicasByStaticWeight(clusters []*clusterv1alpha1.Cluster, weightLis
 	return targetClusters, nil
 }
 
+// getDesireReplicaInfos calculate desireReplicaInfos via the faster algorithm
+// based on the size relationship between remainReplicas and the length of desireReplicaInfos
+func getDesireReplicaInfos(remainReplicas int, desireReplicaInfos, matchClusters map[string]int64) {
```

Comment:
```go
func divideRemainingReplicas(remainingReplicas int, matchedClusterWeights, desiredReplicaInfos map[string]int64) {
	if remainingReplicas <= 0 {
		return
	}

	clusterWeights := helper.SortClusterByWeight(matchedClusterWeights)
	clusterSize := len(clusterWeights)

	if remainingReplicas < clusterSize {
		for i := 0; i < remainingReplicas; i++ {
			desiredReplicaInfos[clusterWeights[i].ClusterName]++
		}
	} else {
		avg, residue := remainingReplicas / clusterSize, remainingReplicas % clusterSize
		for i := 0; i < clusterSize; i++ {
			if i < residue {
				desiredReplicaInfos[clusterWeights[i].ClusterName] = int64(avg) + 1
			} else {
				desiredReplicaInfos[clusterWeights[i].ClusterName] = int64(avg)
			}
		}
	}
}
```

Reply:
Hi, I have finished the logic and added a unit test.